src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -141,6 +137,31 @@ export const isKeyManagerWithProvider = (
       ].includes(keyManager),
   )
 
+export const getMaybeProvider = async (
+  localWalletType: KeyManager | null,
+): Promise<InitialState['provider']> => {
+  if (!localWalletType) return null
+  if (!isKeyManagerWithProvider(localWalletType)) return null
+
+  if (localWalletType === KeyManager.MetaMask) {
+    return (await detectEthereumProvider()) as MetaMaskLikeProvider
+  }
+  if (localWalletType === KeyManager.XDefi) {
+    try {
+      return globalThis?.xfi?.ethereum as unknown as MetaMaskLikeProvider
+    } catch (error) {
+      throw new Error('walletProvider.xdefi.errors.connectFailure')
```

Comment:
it could be handy to console.error the original so we can use the trace for debuggin

Reply:
Might as well 🧠 3f44ff546bda976f17ef2c008944118a296aef9